apps/omg_eth/lib/omg_eth/ethereum_height_monitor.ex

```diff
@@ -163,14 +160,15 @@ defmodule OMG.Eth.EthereumHeightMonitor do
     end
   end
 
-  @spec broadcast_on_new_height(module(), non_neg_integer() | :error) :: :ok | {:error, term()}
-  defp broadcast_on_new_height(_event_bus, :error), do: :ok
+  @spec broadcast_on_new_height(non_neg_integer() | :error) :: :ok | {:error, term()}
+  defp broadcast_on_new_height(:error), do: :ok
 
   # we need to publish every height we fetched so that we can re-examine blocks in case of re-orgs
   # clients subscribed to this topic need to be aware of that and if a block number repeats,
   # it needs to re-write logs, for example
-  defp broadcast_on_new_height(event_bus, height) do
-    apply(event_bus, :broadcast, ["ethereum_new_height", {:ethereum_new_height, height}])
+  defp broadcast_on_new_height(height) do
+    event = OMG.Bus.Event.root_chain_event("ethereum_new_height", :ethereum_new_height, height)
+    OMG.Bus.broadcast(event)
```

Comment:
Why did we follow the way of providing event_bus module as an argument? Behind the scenes is almost surely the `OMG.Bus`, it exactly matches `OMG.Bus` API and we do not mock it in tests.

Reply:
Or it might be the opposite: we should pass the `bus module` as a param everywhere and mock more on tests as we're moving from integration tests to unit + cabbage

WDYT @InoMurko ❓

Reply:
yeah that would be better @pnowosie

Reply:
I brought back `event_bus` module as a state component. If we want to follow that path and pass it as param everywhere, let's do it in a separate PR.

Reply:
the question about mocking a particular component of a system (like the bus) is really just a question about ... how difficult it is to test? In the case of omg bus, it's not. the bus is already a highly decoupled component... and as long as we keep it that way, there's no need for mocking.